src/WebJobs.Extensions.DurableTask/DurableActivityContext.cs

```diff
@@ -102,6 +103,31 @@ internal object GetInput(Type destinationType)
             }
 
             string serializedValue = jToken.ToString(Formatting.None);
+
+            // Object inputs for out-of-proc activities are passed in their JSON-stringified form
+            // with a destination type of System.String. Unfortunately, deserializing a JSON string
+            // to a string causes MessagePayloadDataConverter to throw an exception. This is a
+            // workaround for that case.
+            if (destinationType.Equals(typeof(string)))
+            {
+                // Json.NET has no TryParse or IsValidJson method, so we'll optimistically parse
+                // and catch any exceptions (which shouldn't occur: all (in-proc; JSON and not JSON)
+                // or all other (out-of-proc: not-JSON) inputs with destination System.String should
+                // cast to valid JValues and be handled above.)
+                try
+                {
+                    var deserializedValue = MessagePayloadDataConverter.Default.Deserialize(serializedValue, typeof(object));

```

Comment:
Interesting. Technically, a `string` is also an `object`, so does this deserialization work as intended for all cases?

Reply:
It does - a valid JSON string will parse to a JObject or JArray when the Deserialize() call's objectType is object, but will cause an exception to be thrown when the objectType is string.